src/App.js

```diff
@@ -1,49 +1,52 @@
 import "./styles.css";
-import { Routes, Route } from "react-router-dom";
+import { Routes, Route, useLocation } from "react-router-dom";
 import Mockman from "mockman-js";
-import { Home, Videos, LikedVideos, WatchLater, History, Playlist, SinglePlaylist } from "./pages";
-import { useAsyncFetch, useLogin } from "./hooks";
-import { Navbar } from "./components";
+import { Home, Videos, LikedVideos, WatchLater, History, Playlist, SinglePlaylist, Page404 } from "./pages";
+import { useLogin } from "./hooks";
+import { Navbar, Sidebar } from "./components";
 import { useEffect } from "react";
-import { getLikedVideos, getWatchHistory, getWatchLaterVideos } from "./utils";
-import { useVideos } from "./context/video-context";
+import { useVideo, useLike, useHistory, usePlaylist, useWatchLater } from "./context";
 
 function App() {
-  const {dispatch} = useVideos();
-  
-  useAsyncFetch({
-    url: "/api/videos",
-    dispatchType:"SET_VIDEOS",
-    dispatchPayload:"videos"
-  })
-  
-  useAsyncFetch({
-    url: "/api/categories",
-    dispatchType:"SET_CATEGORIES",
-    dispatchPayload:"categories"
-  })
-  
+  const {getVideos, getCategories} = useVideo();
+  const {getLikedVideos} = useLike();
+  const {getWatchLaterVideos} = useWatchLater();
+  const {getWatchHistory} = useHistory();
+  const {getPlaylists} = usePlaylist();
+  const location = useLocation();
+
   useLogin();
 
   useEffect(()=>{
-    getLikedVideos(dispatch);
-    getWatchLaterVideos(dispatch);
-    getWatchHistory(dispatch);
+    getVideos();
+    getCategories();
+    getLikedVideos();
```

Comment:
Query: Any particular reason for fetching the data at the initial app render and not on individual page render?

Reply:
Yes, it will only load once. I don't want to fetch data everytime the page is opened.